addons/l10n_in_edi_ewaybill/models/account_edi_format.py

```diff
@@ -0,0 +1,601 @@
+                })
+            else:
+                json_payload.update({
+                    "actToStateCode": int(ship_to_details.state_id.l10n_in_tin),
+                    "toPincode": int(extract_digits(ship_to_details.zip)),
+                })
+
+        if invoices.l10n_in_mode == "0":
+            json_payload.update({
+                "transporterId": invoices.l10n_in_transporter_id.vat or "",
+                "transporterName": invoices.l10n_in_transporter_id.name or "",
+            })
+        if invoices.l10n_in_mode in ("2", "3", "4"):
+            json_payload.update({
+                "transMode": invoices.l10n_in_mode,
+                "transDocNo": invoices.l10n_in_transporter_doc_no or "",
+                "transDocDate": invoices.l10n_in_transporter_doc_date and
+                    invoices.l10n_in_transporter_doc_date.strftime("%d/%m/%Y") or "",
+            })
+        if invoices.l10n_in_mode == "1":
+            json_payload.update({
+                "transMode": invoices.l10n_in_mode,
+                "vehicleNo": invoices.l10n_in_vehicle_no or "",
+                "vehicleType": invoices.l10n_in_vehicle_type or "",
+            })
+        return json_payload
+
+    def _get_l10n_in_edi_ewaybill_line_details(self, line, line_tax_details, sign):
+        extract_digits = self._l10n_in_edi_extract_digits
+        tax_details_by_code = self._get_l10n_in_tax_details_by_line_code(
+            line_tax_details.get("tax_details", {}))
+        line_details = {
+            "productName": line.product_id.name,
+            "hsnCode": extract_digits(line.product_id.l10n_in_hsn_code),
+            "productDesc": line.name,
+            "quantity": line.quantity,
+            "qtyUnit": line.product_id.uom_id.l10n_in_code and line.product_id.uom_id.l10n_in_code.split("-")[0] or "OTH",
+            "taxableAmount": self._l10n_in_round_value(line.balance * sign),
+        }
+        if tax_details_by_code.get("igst_rate"):
+            line_details.update({"igstRate": self._l10n_in_round_value(tax_details_by_code["igst_rate"])})
+        else:
+            line_details.update({
+                "cgstRate": self._l10n_in_round_value(tax_details_by_code.get("cgst_rate", 0.00)),
+                "sgstRate": self._l10n_in_round_value(tax_details_by_code.get("sgst_rate", 0.00)),
+            })
+        if tax_details_by_code.get("cess_rate"):
+            line_details.update({"cessRate": self._l10n_in_round_value(tax_details_by_code.get("cess_rate"))})
+        return line_details
+
+    #================================ E-invoice API methods ===========================
+
+    @api.model
+    def _l10n_in_edi_irn_ewaybill_generate(self, company, json_payload):
+        # IRN is created by E-invoice API call so waiting for it.
+        if not json_payload.get("Irn"):
+            return {"error": [{
+                "code": "waiting",
+                "message": _("waiting For IRN generation To create E-waybill")}
```

Comment:
This is good that here you are waiting...

Reply:
still, user can do same time if it's ticked in the journal